src/fastoad/cmd/api.py

```diff
@@ -221,10 +221,20 @@ def list_modules(
     if out is None:
         out = sys.stdout
 
-    if configuration_file_path:
-        conf = FASTOADProblemConfigurator(configuration_file_path)
-        conf._set_configuration_modifier(_PROBLEM_CONFIGURATOR)
-    # As the problem has been configured, BundleLoader now knows additional registered systems
+    if source_path:
+        if isinstance(source_path, str) and pth.isfile(source_path):
+            conf = FASTOADProblemConfigurator(source_path)
+            conf._set_configuration_modifier(_PROBLEM_CONFIGURATOR)
+            # As the problem has been configured,
+            # BundleLoader now knows additional registered systems
+        elif isinstance(source_path, str) and pth.isdir(source_path):
+            RegisterOpenMDAOSystem.explore_folder(source_path)
+        elif any(pth.isdir(x) for x in source_path):
+            for folder_path in source_path:
+                if not pth.exists(folder_path):
+                    _LOGGER.warning("SKIPPED %s: it does not exist.", folder_path)
+                else:
+                    RegisterOpenMDAOSystem.explore_folder(folder_path)
```

Comment:
In your code, there is a first check than the list contains a folder path, then a check on each item that it is an existent path. I see 3 problems here:
- there is an assumption that, at this point, `source_path` is a list
- `pth.isdir` checks path existence, so by checking it again in the `for` loop, the job is done twice
- as the second check is only for existence, if `source_path` contains an existing folder path AND an existing file path, `explore_folder` will be fed also with the file path... and that should be bad.

Therefore I suggest:
```suggestion
        elif isinstance(source_path, list):
            for folder_path in source_path:
                if not pth.isdir(folder_path):
                    _LOGGER.warning("SKIPPED %s: folder does not exist.", folder_path)
                else:
                    RegisterOpenMDAOSystem.explore_folder(folder_path)
        else:
            raise RuntimeError("Unexpected type for source_path")
```

Here again, I end with a `else:`, but I did not put much effort in the error handling, as I find this error unlikely.

Reply:
Thanks for the suggestions! This new `else` means we do not allow to not provide any source_path anymore `source_path=None`.
Do you think we should keep it like that?

Reply:
You got a point. I forgot we need to keep the possibility to have `None `as `source_path`.
Changing `else:` by `elif source_path is not None:` should do the trick, shouldn't it ?

Reply:
I agree! I would also keep the `else` to avoid unexpected types of entries. Do you agree?

Reply:
Actually, my suggestion stand for the `else:` because the only unprocessed case will be that source_path is None, and we want to do nothing in this case.

If you prefer to do it by the book (and I can agree with that), you may replace the `else:` by
```python
elif source_path is None:
    pass # nothing needs to be done
else:
```

This is equivalent to my previous suggestion but more expressive.

Reply:
I agree that is why in the mean time I deleted my comment :)

Reply:
Corrected in ff59930